### PR TITLE
Preserve model mappings for shared AI providers

### LINKED ISSUE #3732: model-mapper config overwritten when multiple upstreams share the same provider
## Bug Description

When an AI route has multiple upstreams pointing to the same provider with different `modelMapping`, the model-mapper WasmPlugin config is silently corrupted — the last upstream's mapping overwrites all others, regardless of weight.

## Root Cause

In `higress-console` backend `AiRouteServiceImpl.writeModelMappingResources()`, upstreams of the same provider resolve to the same service, sharing one model-mapper plugin instance. The method iterates per-upstream and replaces the entire `modelMapping` map each time, causing:

1. **Weight ignored**: weight-0 upstream's mapping can override weight-100's
2. **Non-conflicting keys lost**: only the last upstream's keys survive
3. **Empty mapping deletes siblings**: one upstream with null mapping deletes others' mappings

## Reproduce

1. Create AI route: upstream A (same provider, weight=100, `*→kimi-k2.5`) + upstream B (weight=0, `*→glm-5.1`)
2. model-mapper gets `*: glm-5.1` instead of `*: kimi-k2.5`

## Fix

PR submitted to higress-console: https://github.com/higress-group/higress-console/pull/696

Groups upstreams by service, merges modelMappings (higher weight wins on conflicts), writes once per service.

### PR DESCRIPTION
**Pull request by Aias00**
## Summary
- Group AI route upstreams by resolved service target before writing model-mapper instances.
- Merge modelMapping entries once per service, with higher-weight upstreams taking precedence on conflicts.
- Add a regression test for shared-provider upstreams so empty sibling mappings no longer delete valid mappings.

## Test Plan
- [x] ./mvnw -pl sdk -Dtest=AiRouteServiceTest test
- [x] ./mvnw -pl sdk test

Fixes higress-group/higress#3732

### CODE REVIEW

File: backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java [hunk 1]
```diff
@@ -99,4 +124,49 @@ public void responseCodeIs4xxAnd5xxTest() throws Exception {
...
+		verify(wasmPluginInstanceService, times(1)).addOrUpdate(instanceCaptor.capture());
+		verify(wasmPluginInstanceService, never()).delete(expectedTargets, BuiltInPluginName.MODEL_MAPPER, true);
+
+		WasmPluginInstance instance = instanceCaptor.getValue();
+		Assertions.assertEquals(expectedTargets, instance.getTargets());
+		Assertions.assertTrue(instance.getEnabled());
+		Assertions.assertTrue(instance.getInternal());
+		Assertions.assertEquals(MapUtil.of("*", "high-weight-model", "glm", "glm-model"),
+				instance.getConfigurations().get(ModelMapperConfig.MODEL_MAPPING));
+	}
```
Comment (CH3CHO) [inline backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java:167]: 🟡 **Suggestion: Add a symmetric test for the delete branch.**

The whole point of this fix is that the `delete` call should only happen when *all* upstreams sharing a service have empty mappings, not when *any* of them does. The current test only verifies the `never().delete(...)` branch (at least one non-empty mapping), but doesn't cover the case where every mapping is empty and `delete` should be called exactly once. Without it, a future refactor could silently regress the delete path.

Example additional test (in the same file):

```java
@Test
public void modelMappingDeletedWhenAllUpstreamsForServiceAreEmpty() throws Exception {
    // ... same mock setup as above, but every upstream has Collections.emptyMap()
    // expect: verify(wasmPluginInstanceService, never()).addOrUpdate(any(WasmPluginInstance.class));
    //         verify(wasmPluginInstanceService, times(1)).delete(expectedTargets, BuiltInPluginName.MODEL_MAPPER, true);
}
```

**Severity**: MEDIUM

File: backend/sdk/src/main/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceImpl.java [hunk 2]
```diff
@@ -348,13 +350,30 @@ private void writeModelMappingResources(String routeName, List<AiUpstream> upstr
...
+            targetsByService.putIfAbsent(upstreamService.getName(), targets);
+            upstreamsByService.computeIfAbsent(upstreamService.getName(), k -> new ArrayList<>()).add(upstream);
+        }
+
+        for (Map.Entry<String, List<AiUpstream>> entry : upstreamsByService.entrySet()) {
+            Map<WasmPluginInstanceScope, String> targets = targetsByService.get(entry.getKey());
+            Map<String, String> modelMapping = new HashMap<>();
+            entry.getValue()
+                .stream()
+                .sorted(Comparator.comparingInt(upstream -> Optional.ofNullable(upstream.getWeight()).orElse(0)))
```
Comment (CH3CHO) [inline backend/sdk/src/main/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceImpl.java:370]: 🟢 **Suggestion: Add a comment explaining the sort direction.**

The code sorts ascending by weight, but the intent (per the PR description) is "higher-weight upstreams taking precedence". This works because `putAll` overwrites earlier values, but a reader has to trace through the logic to verify. A short comment would prevent future confusion:

```java
// Sort ascending so higher-weight entries overwrite lower-weight ones via putAll.
.sorted(Comparator.comparingInt(upstream -> Optional.ofNullable(upstream.getWeight()).orElse(0)))
```

**Severity**: LOW

File: backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java [hunk 3]
```diff
@@ -99,4 +124,49 @@ public void responseCodeIs4xxAnd5xxTest() throws Exception {
...
+						.modelMapping(MapUtil.of("*", "high-weight-model")).build(),
+				AiUpstream.builder().provider("shared-provider").weight(0)
+						.modelMapping(MapUtil.of("*", "low-weight-model", "glm", "glm-model")).build(),
+				AiUpstream.builder().provider("shared-provider").weight(200).modelMapping(Collections.emptyMap())
+						.build());
+		Map<WasmPluginInstanceScope, String> expectedTargets =
+				MapUtil.of(WasmPluginInstanceScope.ROUTE, "ai-route-test.internal", WasmPluginInstanceScope.SERVICE,
+						"shared-provider.dns");
+
+		Method method = AiRouteServiceImpl.class.getDeclaredMethod("writeModelMappingResources", String.class,
```
Comment (CH3CHO) [inline backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java]: 🟢 **Suggestion: Drop reflection — change the method to package-private.**

The test uses `Method` reflection to invoke the private `writeModelMappingResources`. A simpler alternative is to drop `private` and let the test call it directly. This is a common Java convention for "package-internal API for tests" and is less brittle than reflection.

In `AiRouteServiceImpl.java` line 345:

```diff
-    private void writeModelMappingResources(String routeName, List<AiUpstream> upstreams) {
+    void writeModelMappingResources(String routeName, List<AiUpstream> upstreams) {
```

In the test (lines 155-158):

```diff
-        Method method = AiRouteServiceImpl.class.getDeclaredMethod("writeModelMappingResources", String.class, List.class);
-        method.setAccessible(true);
-        method.invoke(service, "ai-route-test.internal", upstreams);
+        service.writeModelMappingResources("ai-route-test.internal", upstreams);
```

This also lets you drop the `java.lang.reflect.Method` import added at line 34.

**Severity**: LOW

File: backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java (hunk 1 above)
Comment (Aias00) [reply backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java:167]: Done. Added `modelMappingDeletedWhenAllUpstreamsForServiceAreEmpty`, which verifies `addOrUpdate` is never called and `delete` is called exactly once when all mappings for the shared service are empty. `AiRouteServiceTest` passes with 7 tests.

File: backend/sdk/src/main/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceImpl.java (hunk 2 above)
Comment (Aias00) [reply backend/sdk/src/main/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceImpl.java:370]: Done. Added a comment explaining that ascending order allows higher-weight mappings to overwrite lower-weight entries via `putAll`.

File: backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java (hunk 3 above)
Comment (Aias00) [reply backend/sdk/src/test/java/com/alibaba/higress/sdk/service/ai/AiRouteServiceTest.java]: Done. Made `writeModelMappingResources` package-private, call it directly from the test, and removed the reflection import and checked exception.
